service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/FurtherEvidenceUploadedEventHandler.java

```diff
@@ -32,34 +38,76 @@ public void handleDocumentUploadedEvent(final FurtherEvidenceUploadedEvent event
+                newNonConfidentialDocuments);
+        }
+    }
+
+
+    private List<String> getNewNonConfidentialBundleNames(CaseData caseData, CaseData caseDataBefore,
+                                                          DocumentUploadNotificationUserType userType) {
+
+        List<Element<SupportingEvidenceBundle>> newBundle = getEvidenceBundle(caseData, userType);
+        List<Element<SupportingEvidenceBundle>> oldBundle = getEvidenceBundle(caseDataBefore, userType);
+
+        List<String> documentNames = new ArrayList<String>();
+
+        unwrapElements(newBundle).forEach(newDoc -> {
+            if (!newDoc.isConfidentialDocument() && !unwrapElements(oldBundle).contains(newDoc)) {
+                documentNames.add(newDoc.getName());
+            }
+        });
+        return documentNames;
+    }
+
+    private List<Element<SupportingEvidenceBundle>> getEvidenceBundle(CaseData caseData,
+                                                                      DocumentUploadNotificationUserType userType) {
+        if (userType == LOCAL_AUTHORITY) {
+            return caseData.getFurtherEvidenceDocumentsLA();
+        } else if (userType == SOLICITOR) {
+            List<Element<SupportingEvidenceBundle>> furtherEvidenceBundle =
+                caseData.getFurtherEvidenceDocumentsSolicitor();
+            List<Element<SupportingEvidenceBundle>> respondentStatementsBundle =
+                getEvidenceBundleFromRespondentStatements(caseData);
+
+            return concatEvidenceBundles(furtherEvidenceBundle, respondentStatementsBundle);
+        } else {
+            return caseData.getFurtherEvidenceDocuments();
         }
     }
 
-    private static boolean hasNewNonConfidentialDocuments(List<Element<SupportingEvidenceBundle>> newEvidenceBundle,
-                                                          List<Element<SupportingEvidenceBundle>> oldEvidenceBundle) {
-        List<SupportingEvidenceBundle> oldEvidenceBundleUnwrapped = unwrapElements(oldEvidenceBundle);
-        return unwrapElements(newEvidenceBundle).stream()
-            .anyMatch(d -> oldEvidenceBundleUnwrapped.stream()
-                .noneMatch(old -> old.getDocument().equals(d.getDocument()))
-                && !d.isConfidentialDocument());
+    private List<Element<SupportingEvidenceBundle>> getEvidenceBundleFromRespondentStatements(CaseData caseData) {
+        List<Element<SupportingEvidenceBundle>> evidenceBundle = new ArrayList<Element<SupportingEvidenceBundle>>();
+        caseData.getRespondentStatements().forEach(statement -> {
+            evidenceBundle.addAll(statement.getValue().getSupportingEvidenceBundle());
+        });
+        return evidenceBundle;
+    }
+
+    private List<Element<SupportingEvidenceBundle>> concatEvidenceBundles(List<Element<SupportingEvidenceBundle>> b1,
+                                                                          List<Element<SupportingEvidenceBundle>> b2) {
+        if (b1 == null) {
```

Comment:
how does this handle both being `null`, it would hit this block and return `b2` which would be `null`. Would you want to return `null`? I know that the `unwrapElements` method covers for this but was just wondering if it was intentional?

Reply:
if you wanted to never return null the first to return statements could be `defaultIfNull(b2, List.of())`

Reply:
Good spot! I don't think it's possible for them both to be null – they wouldn't be able to submit without uploading a document... But as it is possible in the code - so I will add a `defaultIfNull`! Thank you